app/reducers/pages.js

```diff
@@ -68,3 +70,55 @@ export const selectGroupsForHierarchy = createSelector(
     }))
   })
 );
+
+export const selectPageHierarchy = createSelector(
+  (state, props) => props.sections,
+  state => state,
+  (sections, state) =>
+    Object.keys(sections).map(sectionKey =>
+      sections[sectionKey].hierarchySectionSelector(state, {
+        title: sections[sectionKey].title
+      })
+    )
+);
+
+export const selectFlatpageForPages = createSelector(
+  selectPageBySlug,
+  (state, props) => props.pageSlug,
+  (selectedPage, pageSlug) => ({
+    selectedPage,
+    selectedPageInfo: {
+      actionGrant: selectedPage.actionGrant || [],
+      title: selectedPage.title,
+      editUrl: `/pages/info/${pageSlug}/edit`
+    }
+  })
+);
+
+export const selectComiteeForPages = createSelector(
+  (state, props) => selectGroup(state, { groupId: Number(props.pageSlug) }),
+  (state, props) =>
+    selectMembershipsForGroup(state, { groupId: Number(props.pageSlug) }),
+  (group, memberships) => {
+    const selectedPageInfo = group && {
+      actionGrant: group.actionGrant || [],
+      title: group.name,
+      editUrl: `/admin/groups/${group.id}/settings`
+    };
+    return {
+      selectedPage: group && { ...group, memberships },
```

Comment:
this will make selectedPage and selectedPageInfo be `false` if group doesn't exist - which seems a bit weird?

Reply:
ah, didn't realize you just moved this logic from another file - feel free to look at in another PR

Reply:
But i wrote the logic (the night before launch) the first time, so i should fix it... :laughing:

Reply:
But after some testing, it turns out that 
```js
const group = undefined;
group && {...group, memberships} // Evaluates to undefined, not false
```